ksqldb-serde/src/test/java/io/confluent/ksql/serde/avro/AvroDataTranslatorTest.java

```diff
@@ -288,4 +292,37 @@ public void shouldDropOptionalFromRootStructSchema() {
     // Then:
     assertThat("Root required", translator.getAvroCompatibleSchema().isOptional(), is(false));
   }
+
+  @Test
+  public void shouldRejectUnmatchingDecimalSchema() {
+    // Given:
+    final Schema ksqlSchema = DecimalUtil.builder(4, 2).build();
+    final Schema topicSchema = DecimalUtil.builder(6, 3).build();
+
+    // When:
+    final AvroDataTranslator translator =
+        new AvroDataTranslator(ksqlSchema, AvroProperties.DEFAULT_AVRO_SCHEMA_FULL_NAME);
+
+    // Then:
+    final ArithmeticException e = assertThrows(
+        ArithmeticException.class,
+        () -> translator.toKsqlRow(topicSchema, new BigDecimal("123.456")));
+    assertThat(e.getMessage(), containsString("A field with precision 4 and scale 2 must round"));
+  }
+
+  @Test
+  public void shouldForceUnmatchingDecimalSchemaIfPossible() {
```

Comment:
Maybe a dumb question, could we support a conversion from 1234.5 to a 5 precision and 0 scale decimal number? Are there any other invalid cases we could test out?

Reply:
No, we don't support rounding for conversions now. Added a test for that case.